cfscrape/__init__.py

```diff
@@ -179,55 +202,68 @@ def solve_cf_challenge(self, resp, **original_kwargs):
 
         # Solve the Javascript challenge
         answer, delay = self.solve_challenge(body, domain)
-        params["jschl_answer"] = answer
+        if method == 'POST':
+            cloudflare_kwargs["data"]["jschl_answer"] = answer
+        elif method == 'GET':
+            cloudflare_kwargs["params"]["jschl_answer"] = answer
 
         # Requests transforms any request into a GET after a redirect,
         # so the redirect has to be handled manually here to allow for
         # performing other types of requests even as the first request.
-        method = resp.request.method
         cloudflare_kwargs["allow_redirects"] = False
 
         # Cloudflare requires a delay before solving the challenge
         time.sleep(max(delay - (time.time() - start_time), 0))
 
         # Send the challenge response and handle the redirect manually
         redirect = self.request(method, submit_url, **cloudflare_kwargs)
-        redirect_location = urlparse(redirect.headers["Location"])
-
-        if not redirect_location.netloc:
-            redirect_url = urlunparse(
-                (
-                    parsed_url.scheme,
-                    domain,
-                    redirect_location.path,
-                    redirect_location.params,
-                    redirect_location.query,
-                    redirect_location.fragment,
+        if "Location" in redirect.headers:
+            redirect_location = urlparse(redirect.headers["Location"])
+
+            if not redirect_location.netloc:
+                redirect_url = urlunparse(
+                    (
+                        parsed_url.scheme,
+                        domain,
+                        redirect_location.path,
+                        redirect_location.params,
+                        redirect_location.query,
+                        redirect_location.fragment,
+                    )
                 )
-            )
-            return self.request(method, redirect_url, **original_kwargs)
-        return self.request(method, redirect.headers["Location"], **original_kwargs)
+                return self.request(method, redirect_url, **original_kwargs)
+            return self.request(method, redirect.headers["Location"], **original_kwargs)
+        elif 'cf_clearance' in redirect.headers['Set-Cookie']:
```

Comment:
@alzamer2 could you add small check if 'Set-Cookie' exists in redirect.headers
`elif "Set-Cookie" in redirect.headers and "cf_clearance" in redirect.headers["Set-Cookie"]:`

Reply:
hmm sorry for delay, but after checking i did not understand why this change, do you have an error and this change fix it? if so then this change will make more error
can you try this and see if it fix your error?
`elif "Set-Cookie" in redirect.headers:`
`    if 'cf_clearance' in redirect.headers['Set-Cookie']:`
`        resp = self.request(self.org_method, submit_url, cookies = redirect.cookies)`
`        return resp`
`    else:`
`        return self.request(method, redirect_url, **original_kwargs)`

Reply:
I tested it out on https://www.realcccam.com/freecccam.php, https://www.startupranking.com/top/0/3
`
import requests
import cfscrape

url = "https://www.realcccam.com/freecccam.php"
tokens, user_agent = cfscrape.get_tokens(url)
print(requests.get(url, cookies=dict(tokens), headers={"User-Agent": user_agent}))
`

Reply:
what i'm trying to say that if these is error from 
`elif 'cf_clearance' in redirect.headers['Set-Cookie']:`
then 
`elif "Set-Cookie" in redirect.headers and "cf_clearance" in redirect.headers["Set-Cookie"]:`
will return same error
because test will check `redirect.headers["Set-Cookie"]` anyway 
that why its better to add `elif "Set-Cookie" in redirect.headers` in line befoer `if 'cf_clearance' in redirect.headers['Set-Cookie']:`

Reply:
ok @grmnz
i updated the code and added test to check if "Set-Cookie" in redirect.headers